src/TreeHouse/SnapshotStore/Snapshot.php

```diff
@@ -18,21 +21,35 @@
     private $data;
 
     /**
+     * @var string
+     */
+    private $checksum;
+
+    /**
+     * @var string
+     */
+    private $class;
+
+    /**
      * @var \DateTime
      */
     private $datetimeCreated;
 
     /**
-     * @param $aggregateId
+     * @param string $aggregateId
      * @param int $aggregateVersion
      * @param array $data
+     * @param string $class
+     * @param string $checksum
      */
-    public function __construct($aggregateId, $aggregateVersion, array $data)
+    public function __construct($aggregateId, $aggregateVersion, array $data, $checksum, $class)
```

Comment:
Combined with the strict `string` return of `getCheksum` and `getClass` you're asking for trouble here. I'd drop the strict returns or more preferably; make the constructor strict.